src/audio/mux/mux.c

```diff
@@ -178,6 +178,31 @@ static int mux_verify_params(struct comp_dev *dev,
 	return 0;
 }
 
+/* set component audio stream parameters */
+static int demux_params(struct comp_dev *dev,
+			struct sof_ipc_stream_params *params)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	struct comp_buffer *sourceb;
+	int err;
+
+	comp_info(dev, "demux_params()");
+
+	err = mux_verify_params(dev, params);
+	if (err < 0) {
+		comp_err(dev, "demux_fir_params(): pcm params verification failed.");
+		return -EINVAL;
+	}
+
+	sourceb = list_first_item(&dev->bsource_list, struct comp_buffer,
+				  sink_list);
```

Comment:
@keyonjie the change looks correct but to avoid duplication of code, can we re-use the same mux_params  and check for dev->drv->type and set the buffer to sourcebuffer or sink_buffer accordingly?

Reply:
@ranj063 we can for sure making change like that, but that is actually another kind of waste, we already know it is mux or demux, why we need to check it again?

Reply:
Can we have an inline comment here describing what we are doing with channels and frame format here.

Reply:
@keyonjie what I was suggesting to avoid unnecessary code duplication. It is true that we already know it's a mux or a demux, then we can also argue to separate these into 2 files, no? This change really is about whether we're getting the params from the source or the sink buffer. So adding an extra line to add the check is a lot less wasteful that duplicating the entire params function

Reply:
> @keyonjie what I was suggesting to avoid unnecessary code duplication. It is true that we already know it's a mux or a demux, then we can also argue to separate these into 2 files, no? This change really is about whether we're getting the params from the source or the sink buffer. So adding an extra line to add the check is a lot less wasteful that duplicating the entire params function

Fair enough, let me change it, thanks.

Reply:
> Can we have an inline comment here describing what we are doing with channels and frame format here.

I had difficulty to understand it at first glance also, let me add some comment here.